File: basemodule/src/main/java/com/android/base/mmodel/BaseResponse.java
package com.android.base.mmodel;

public class BaseResponse<T> {

    public static final int SUCCESS = 0;
    public static final int FAIL = 1;

    /**
     * 0：成功，1：失败
     */
    public int errorCode;

    public String errorMsg;

    public T data;

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

}

File: basemodule/src/main/java/com/android/base/mmodel/BaseResponse.kt
package com.android.base.mmodel

class BaseResponse<T: BaseData> {

    /**
     * 0：成功，1：失败
     */
    var errorCode: Int = 0

    lateinit var errorMsg: String

    lateinit var data: T

    companion object {

        val SUCCESS = 0
        val FAIL = 1
    }

}